src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/ConsoleDebugTargetsProvider.cs

```diff
@@ -335,6 +338,14 @@ private static bool IsRunProjectCommand(ILaunchProfile profile)
             return new Tuple<string, string, string>(runCommand, runArguments, runWorkingDirectory);
         }
 
+        private async Task<string> GetOutputDirectoryAsync(ConfiguredProject configuredProject)
+        {
+            var properties = configuredProject.Services.ProjectPropertiesProvider.GetCommonProperties();
+            var outdir = await properties.GetEvaluatedPropertyValueAsync("OutDir").ConfigureAwait(false);
```

Comment:
Are we safe to rely on `OutDir` in all cases?

Reply:
Is this still an issue https://github.com/Microsoft/msbuild/issues/87 (https://mfreidge.wordpress.com/2017/01/22/outdir-vs-outputpath-for-visual-studio-2015-and-asp-net-core/)?

Reply:
Tagging @rainersigwald for an opinion.

Reply:
Definitely no **guarantee**, unfortunately: even without the ASP.NET confusion, a project (or NuGet package) could customize outputs so they don't respond to the convention.

That said, ASP.NET Core projects look to me like they behave ok, so something must have changed since that blog post.

I think this design is safe for this, but see Nat's other comment too for belt + suspenders.

Reply:
I push another change which adds the fallback logic suggested by Rainer